nixos/modules/services/web-servers/darkhttpd.nix

```diff
@@ -0,0 +1,85 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.darkhttpd;
+
+  args = concatStringsSep " " ([
+    cfg.rootDir
+    "--port ${toString cfg.port}"
+    "--addr ${cfg.address}"
+  ] ++ cfg.extraArgs
+    ++ optional cfg.hideServerId             "--no-server-id"
+    ++ optional config.networking.enableIPv6 "--ipv6");
+
+in {
+  options.services.darkhttpd = with types; {
+    enable = mkEnableOption "DarkHTTPd web server";
+
+    port = mkOption {
+      default = 80;
+      type = ints.u16;
+      description = ''
+        Port to listen on.
+        Pass 0 to let the system choose any free port for you.
+      '';
+    };
+
+    address = mkOption {
+      default = "127.0.0.1";
+      type = str;
+      description = ''
+        Address to listen on.
+        Pass `all` to listen on all interfaces.
+      '';
+    };
+
+    package = mkOption {
+      default = pkgs.darkhttpd;
+      type = package;
```

Comment:
I'd actually rather not have this option. In general, the package can already be overridden with overlays, whose only disadvantage is that you can't override the package *only* for this module, it always happens for the whole system. This can be a problem when you e.g. want to override `pkgs.curl`, which has lots of dependents, so it would rebuild a whole lot, which doesn't apply to `darkhttpd`.

Every option increases evaluation time, so I'd rather have this be used for options that allow you to do something you couldn't otherwise do.

Reply:
Also see https://github.com/NixOS/nixpkgs/pull/48289#issuecomment-432283577.

Reply:
Also https://github.com/NixOS/nixpkgs/pull/49228#discussion_r228859491